### PR TITLE
Add ownerReferences for resources created by CR

### PR DESCRIPTION
**Pull request by nmagnezi**
Fixes: #235
Co-authored-by: Fred Rolland <frolland@redhat.com>

### CODE REVIEW

**Comment by rollandf** [comment]
@metalmatze @brancz @kakkoyun @squat PTAL

File: environments/operator/obs-operator.jsonnet [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+
+local addOwner(k, v) = v { metadata+: { ownerReferences: [{
+  apiVersion: config.apiVersion,
+  blockOwnerdeletion: true,
+  controller: true,
+  kind: config.kind,
+  name: config.name,
+  uid: config.uid,
+}] } };
+local ownermanifests = std.mapWithKey(addOwner, manifests);
```
Comment (brancz) [inline environments/operator/obs-operator.jsonnet:14]: as you're not using the key, you can just use `std.map`

File: environments/operator/obs-operator.jsonnet [hunk 2]
```diff
@@ -1,3 +1,18 @@
...
+
+{
+  manifests: ownermanifests,
```
Comment (brancz) [inline environments/operator/obs-operator.jsonnet:17]: I would inline the `std.map` call here, no need for the local variables above except the imports

File: environments/operator/obs-operator.jsonnet [hunk 3]
```diff
@@ -1,3 +1,18 @@
-(import '../../components/observatorium.libsonnet') + {
-  config+:: (import 'operator-config.libsonnet'),
-} + (import '../../components/observatorium-configure.libsonnet')
+local config = import 'operator-config.libsonnet';
+local manifests = ((import '../../components/observatorium.libsonnet') + {
+                     config+:: config,
+                   } + (import '../../components/observatorium-configure.libsonnet')).manifests;
+
+local addOwner(k, v) = v { metadata+: { ownerReferences: [{
```
Comment (brancz) [inline environments/operator/obs-operator.jsonnet:6]: we tend to put fields of nested objects on new lines, so something like:

```jsonnet
... v {
  metadata+: {
    ownerReferences: [{
      ...
    }],
  },
}
```

File: environments/operator/obs-operator.jsonnet (hunk 1 above)
Comment (rollandf) [reply environments/operator/obs-operator.jsonnet:14]: std.map works for array.
This is an object here:
`RUNTIME ERROR: std.map second param must be array / string, got object`

Comment (brancz) [reply environments/operator/obs-operator.jsonnet:14]: ah ok yes, makes sense :+1:

File: environments/operator/obs-operator.jsonnet (hunk 3 above)
Comment (rollandf) [reply environments/operator/obs-operator.jsonnet]: Done

File: environments/operator/obs-operator.jsonnet (hunk 2 above)
Comment (rollandf) [reply environments/operator/obs-operator.jsonnet]: Done

File: environments/operator/obs-operator.jsonnet [hunk 4]
```diff
@@ -1,3 +1,21 @@
...
+      controller: true,
+      kind: config.kind,
+      name: config.name,
+      uid: config.uid,
+    }],
+  },
+};
+
+{
+  manifests: std.mapWithKey(addOwner, manifests),
```
Comment (brancz) [inline environments/operator/obs-operator.jsonnet:20]: I meant to inline the function here

Comment (rollandf) [reply environments/operator/obs-operator.jsonnet]: Done

**Pull request merged**